### DISCUSSION TITLE
Is it possible to pull branch-specific env variables using the CLI?

### DISCUSSION THREAD

**Discussion by piedrahitapablo** [vercel/vercel#8994] [Ideas] [answered]
Hi, I'm currently using the vercel CLI to deploy from github actions using [this](https://vercel.com/guides/how-can-i-use-github-actions-with-vercel) guide. However there's one thing that is not clear to me, in vercel I can setup preview env variables for specific branches, but when I pull the environment from vercel, those branch-specific variables are not pulled and I don't see a command or flag to do it. 

Do you know if I'm missing something? or is that not possible at all right now?

Thanks in advance for your help!

**Comment by piedrahitapablo** [comment, marked as answer]
I just saw that this is a duplicate of #8652 and #8528, but I don't know how to close it. Sorry for posting it.

**Reply by amyegan** [reply]
No problem! I'm glad you asked. The team already created a ticket for this, as you found, but it's always good to know what features people want and what information was difficult to find. I'm adding a link back to this in the ticket. I'll also move this post to the `Idea` category to track as a feature request.

**Reply by piedrahitapablo** [reply]
Thanks @amyegan!

**Reply by amyegan** [reply]
🚀 The `--git-branch` option is available on `vc pull` and `vc env pull` now in [vercel@28.9.0](https://github.com/vercel/vercel/releases/tag/vercel%4028.9.0)!

**Reply by piedrahitapablo** [reply]
Thanks @amyegan
